neo4j/src/test/java/org/hibernate/ogm/datastore/neo4j/test/mapping/CompositeIndexTest.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+
+package org.hibernate.ogm.datastore.neo4j.test.mapping;
+
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.Index;
+import javax.persistence.Table;
+
+import org.hibernate.ogm.utils.TestForIssue;
+
+import org.junit.Test;
+
+/**
+ * @author The Viet Nguyen &amp;ntviet18@gmail.com&amp;
+ */
+@TestForIssue(jiraKey = "OGM-1462")
+public class CompositeIndexTest extends Neo4jJpaTestCase {
+
+	@Entity
+	@Table(indexes = { @Index(columnList = "firstname,lastname"), @Index(columnList = "middlename") })
+	static class Person {
+
+		@Id
+		private String id;
+
+		private String firstname;
+
+		private String middlename;
+
+		private String lastname;
+
+		public String getId() {
+			return id;
+		}
+
+		public void setId(String id) {
+			this.id = id;
+		}
+
+		public String getFirstname() {
+			return firstname;
+		}
+
+		public void setFirstname(String firstname) {
+			this.firstname = firstname;
+		}
+
+		public String getMiddlename() {
+			return middlename;
+		}
+
+		public void setMiddlename(String middlename) {
+			this.middlename = middlename;
+		}
+
+		public String getLastname() {
+			return lastname;
+		}
+
+		public void setLastname(String lastname) {
+			this.lastname = lastname;
+		}
+	}
+
+	@Test
+	public void test() {
+		// TODO: create test plans
```

Comment:
In MongoDB there are already a lot of test cases for this issue that you can copy and adapt.  MongoDB is more powerful in term of indexes you will need to pick the one that make sense for Neo4j.